src/modules/Game.class.js

```diff
@@ -50,19 +64,189 @@ class Game {
+    this.boardScore = 0;
+
+    this.currentStatus = 'idle';
+    this.isGameActive = false;
+    this.isGameWon = false;
+    this.isGameLose = false;
+  }
+
+  placeNewCell() {
+    let rRow, rColumn;
+
+    do {
+      rRow = Math.floor(Math.random() * 4);
+      rColumn = Math.floor(Math.random() * 4);
+    } while (this.board[rRow][rColumn] !== 0);
+
+    this.board[rRow][rColumn] = this.createNewCell();
+  }
+
+  createNewCell() {
+    const randomValue = Math.random();
+
+    return randomValue < 0.1 ? 4 : 2;
+  }
+
+  move(direction) {
+    if (!this.isGameActive) {
+      return;
+    }
+
+    const numCols = this.board[0].length;
+    let currentTable = JSON.parse(JSON.stringify(this.board));
+    let addScore = 0;
+
+    const transpose = (table) => {
+      return table[0].map((_, colIndex) => table.map((row) => row[colIndex]));
+    };
+
+    const reverseRow = (table) => {
+      return table.map((row) => row.slice().reverse());
+    };
 
-  // Add your own methods here
+    const moveTable = (table) => {
+      const newTable = table.map((row) => {
+        let newRow = row.filter((num) => num !== 0);
+        const zeroToAdd = numCols - newRow.length;
+
+        newRow = [...Array(zeroToAdd).fill(0), ...newRow];
+
+        for (let i = newRow.length; i >= 0; i--) {
+          if (newRow[i - 1] === newRow[i]) {
+            newRow[i - 1] *= 2;
+            newRow[i] = 0;
+            addScore += newRow[i - 1];
+            i--;
+          }
+        }
+
+        newRow = newRow.filter((num) => num !== 0);
+
+        const zerosToAddEnd = numCols - newRow.length;
+
+        newRow = [...Array(zerosToAddEnd).fill(0), ...newRow];
+
+        return newRow;
+      });
+
+      return newTable;
+    };
+
+    const isGameOver = () => {
+      const Gamefield = 4;
+
+      for (let i = 0; i < Gamefield; i++) {
+        for (let j = 0; j < Gamefield; j++) {
+          if (this.board[i][j] === 0) {
+            return false;
+          }
+        }
+      }
+
+      for (let i = 0; i < Gamefield; i++) {
+        for (let j = 0; j < Gamefield; j++) {
+          if (j < Gamefield - 1 && this.board[i][j] === this.board[i][j + 1]) {
+            return false;
+          }
+
+          if (i < Gamefield - 1 && this.board[i][j] === this.board[i + 1][j]) {
+            return false;
+          }
+        }
+      }
+
+      this.isGameActive = false;
+      this.isGameLose = true;
+
+      return true;
+    };
+
+    const makeMove = (moveToSide) => {
+      if (JSON.stringify(moveToSide) === JSON.stringify(currentTable)) {
+        return;
+      }
+
+      this.board = moveToSide;
+      currentTable = moveToSide;
+
+      if (this.board.flat().includes(2048)) {
+        this.isGameActive = false;
+        this.isGameWon = true;
+
+        this.getStatus();
+
+        return;
+      }
+
+      this.placeNewCell();
+
+      this.isAbleMove = !isGameOver();
+    };
+
+    switch (direction) {
+      case 'up':
+        const moveUp = transpose(
```

Comment:
`moveUp` sounds like this is a function. It's a bit confusing. Rename it to reflect the value stored in this variable. Same below